specs/agents/tracing-spans-destination.md

```diff
@@ -158,8 +160,16 @@ ES field: [`destination.address`](https://www.elastic.co/guide/en/ecs/current/ec
 
 Address is the destination network address: hostname (e.g. `localhost`), FQDN (e.g. `elastic.co`), IPv4 (e.g. `127.0.0.1`) IPv6 (e.g. `::1`)
 
+Agents MAY offer a public API to set this field so that users can override the automnatically discovered one. 
+This includes the ability to set `null` or empty value in order to override the automatically-set value.
+User-supplied value MUST have the highest precedence, regardless if it was set before or after the automatic setting is invoked.
+
 #### `context.destination.port`
 
 ES field: [`destination.port`](https://www.elastic.co/guide/en/ecs/current/ecs-destination.html#_destination_field_details)
 
 Port is the destination network port (e.g. 443)
+
+Agents MAY offer a public API to set this field so that users can override the automnatically discovered one. 
+This includes the ability to set a non-positive value in order to override the automatically-set value.
```

Comment:
```suggestion
This includes the ability to set a non-positive value in order to unset the automatically-set value.
```

Shouldn't 0 also be treated as unsetting the field?

Reply:
That's why I used `non-positive` 😄

Reply:
You explicitly want to allow negative values to mean "unset"? E.g. `span.setDestinationPort(-42)` or whatever?

Reply:
Since positive integers represent valid TCP port numbers (even thought there is a defined max to this range), non-positive represent invalid numbers, so it seemed the easiest. If someone chooses to use `-42`, I don't think we should think it means anything else other than: "disable this field".

> either do not specially treat negative values 

If by that you mean that when someone makes the effort to set `-42` we should ignore that and use the automatically discovered port, then this definitely sounds like a worse option to me.

> or error on negative values being invalid for a port

This seems like additional logic without additional value.

Since you don't feel strongly about this, I would go for the passive approach and not change the spec 🙂 . But if more people feel we are better off restricting invalid values to `0`, I won't insist either.

Reply:
> I would go for the passive approach 

I'm cool with that. :)